### PR TITLE
Functions Refactor Runtimes

### PR DESCRIPTION
**Pull request by PineappleIOnic**
This PR contains all the new rewritten runtimes, tests and documentation for the new runtime style defined within the runtimes refactor RFC seen here: https://github.com/appwrite/rfc/pull/29 for the new functions refactor.

Notes: Currently we are missing both the Java runtime and ~~the Swift runtime~~ since they are still currently being worked on.

Edit: Swift runtime has been created.

### CODE REVIEW

File: runtimes/dart-2.13/.packages [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# This file is deprecated. Tools should instead consume
```
Comment (lohanidamodar) [inline runtimes/dart-2.13/.packages:1]: This file should be ignored in git

File: runtimes/dart-2.13/README.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+A runtime container for Dart 2.13
+Developed to easily run and build Dart Code within a container.
+
+## Creating a function for the runtime
+To create a function for the runtime, you must create a asyncronous Dart function that returns a `Future<void>`. It must take two parameters, `Request` and `Response` in that order. What you can expect these functions to contain can be found below in the `Custom Structures/Classes` section.
+
+The function **must** be named `start` otherwise it will not compile.
+
+## Handling User Dependencies
+User Dependencies are not currently supported. This is a planned feature.
```
Comment (lohanidamodar) [inline runtimes/dart-2.13/README.md:11]: Let's update this, user dependencies are supported right? we build user's code get their dependencies during build?

File: runtimes/dart-2.13/appwrite_function_types/lib/appwrite_function_types.dart [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import 'dart:convert';
```
Comment (lohanidamodar) [inline runtimes/dart-2.13/appwrite_function_types/lib/appwrite_function_types.dart:1]: These function types for dart and type definition for other languages should be published, as developers will have to use it in their code. Let's discuss this.

File: runtimes/dart-2.14/user_code/analysis_options.yaml [hunk 4]
```diff
@@ -0,0 +1,16 @@
+# Defines a default set of lint rules enforced for projects at Google. For
```
Comment (lohanidamodar) [inline runtimes/dart-2.14/user_code/analysis_options.yaml:1]: We should add Dart 2.15 as well.

**Review by lohanidamodar — CHANGES_REQUESTED** [review]
Went through the Dart runtimes and left some feedback.

File: README.md [hunk 5]
```diff
@@ -69,13 +80,18 @@ First of all, a script for the runtime environment must be created. Plenty of ex
 
 After that start options must be added to the `setUp()` method found in `tests/Runtimes/RuntimesTest.php`.
 
+**WARNING:** On MacOS and Windows the `/tmp/builtCode/` needs to be created before the tests are run. This is due to a wierd permission issue on Docker Desktop that occours if the directory does not exist since Docker Desktop does not create it automatically.
```
[the remark this answers is not in the document]
Comment (PineappleIOnic) [reply README.md:83]: Good Catch, not really sure how I should translate that into windows. I've never really ran tests on windows before.

Maybe we shouldn't mention windows?

File: runtimes/dart-2.12/README.md [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+A runtime container for Dart 2.12
+Developed to easily run and build Dart Code within a container.
+
+## Creating a function for the runtime
+To create a function for the runtime, you must create a asyncronous Dart function that returns a `Future<void>`. It must take two parameters, `Request` and `Response` in that order. What you can expect these functions to contain can be found below in the `Custom Structures/Classes` section.
+
+The function **must** be named `start` otherwise it will not compile.
+
+## Handling User Dependencies
+User Dependencies are not currently supported. This is a planned feature.
```
[the remark this answers is not in the document]
Comment (PineappleIOnic) [reply runtimes/dart-2.12/README.md]: This has been fix with the new dart readmes

File: runtimes/dart-2.12/pubspec.yaml.fallback [hunk 7]
```diff
@@ -0,0 +1,14 @@
+name: appwrite_dart_function
```
[the remark this answers is not in the document]
Comment (PineappleIOnic) [reply runtimes/dart-2.12/pubspec.yaml.fallback:1]: The fallback file? or the name?

**Comment by PineappleIOnic** [comment]
This PR will be closed as these runtimes have been merged into OpenRuntimes

**Pull request closed without merging**